### PR TITLE
Updates machine files for PNNL's constance machine for ACME

### PR DESCRIPTION
**Pull request by singhbalwinder**
This PR adds machine files for Constance machine which is hosted at PNNL
campus. The machine files are added for ACME CIME configuration only.

Test suite: cime_developer, All PASS
Test baseline: none
Test namelist changes: none
Test status: [bit for bit]

User interface changes?: None

Code review:

### CODE REVIEW

**Comment by rljacob** [comment]
@singhbalwinder is this also on the maint-cime5.1 branch?

File: cime_config/acme/machines/config_machines.xml [hunk 1]
```diff
@@ -993,33 +993,100 @@
...
+	   <modules mpilib="openmpi">
+	     <command name="load">openmpi/1.8.3</command>
+	   </modules>
+	   <modules>
+	     <command name="load">netcdf/4.3.2</command>
+	   </modules>
+	 </module_system>
+	 <environment_variables>
+	   <env name="OMP_STACKSIZE">64M</env>
+	   <env name="NETCDF_HOME">$NETCDF_LIB/../</env>
```
Comment (jgfouca) [inline cime_config/acme/machines/config_machines.xml]: I think this needs to be $ENV{NETCDF_LIB}/...

File: cime_config/acme/machines/config_machines.xml [hunk 2]
```diff
@@ -993,33 +993,100 @@
...
+	 </environment_variables>
+	 <environment_variables compiler="intel">
+	   <env name="MKL_PATH">$MLIB_LIB</env>
```
Comment (jgfouca) [inline cime_config/acme/machines/config_machines.xml]: $ENV{MLIB_LIB}

**Comment by singhbalwinder** [comment]
@rljacob : I branched off from the master branch. Is there a different workflow which is recommended?

@jgfouca : I will make and push the changes you asked for. Thanks for reviewing this PR.

**Comment by singhbalwinder** [comment]
I have the fixed the environment variables by adding $ENV.

**Comment by rljacob** [comment]
You need to branch off of maint-cime5.1, or the common ancestor of it and master, to get these changes on the maint-cime5.1 branch.

**Comment by jgfouca** [comment]
@singhbalwinder you should be able to rebase this branch onto maint-cime5.1.

**Comment by singhbalwinder** [comment]
Thanks @jgfouca . I am working on it now.

File: cime_config/acme/machines/config_machines.xml (hunk 1 above)
Comment (singhbalwinder) [inline cime_config/acme/machines/config_machines.xml]: @jgfouca : If I use $ENV{NETCDF_LIB}, the makefile picks up `-I{NETCDF_LIB}/..//include` as include path. For some reason, $ENV is not expanding NETCDF_LIB variable.  Following are the contents of my .env_mach_specific.sh:
module purge
module load perl/5.20.0
module load cmake/3.3.0
module load python/2.7.8
module load intel/15.0.1
module load mkl/15.0.1
module load mvapich2/2.1
module load netcdf/4.3.2
export OMP_STACKSIZE=64M
export NETCDF_HOME=$ENV{NETCDF_LIB}/../
export MKL_PATH=$ENV{MLIB_LIB}


I am a csh user, contents of my .env_mach_specific.csh are:

module purge
module load perl/5.20.0
module load cmake/3.3.0
module load python/2.7.8
module load intel/15.0.1
module load mkl/15.0.1
module load mvapich2/2.1
module load netcdf/4.3.2
setenv OMP_STACKSIZE 64M
setenv NETCDF_HOME $ENV{NETCDF_LIB}/../
setenv MKL_PATH $ENV{MLIB_LIB}

File: cime_config/acme/machines/config_machines.xml [hunk 3]
```diff
@@ -993,33 +993,100 @@
...
+	   <modules>
+	     <command name="load">netcdf/4.3.2</command>
+	   </modules>
+	 </module_system>
+	 <environment_variables>
+	   <env name="OMP_STACKSIZE">64M</env>
+	   <env name="NETCDF_HOME">$ENV{NETCDF_LIB}/../</env>
+	 </environment_variables>
+	 <environment_variables compiler="intel">
+	   <env name="MKL_PATH">$ENV{MLIB_LIB}</env>
```
Comment (jedwards4b) [inline cime_config/acme/machines/config_machines.xml]: Does the MKL_PATH setting work?  Not sure why this would work but not line 1085?

Comment (singhbalwinder) [inline cime_config/acme/machines/config_machines.xml]: The build stops with netcdf error in csm share. I think MKL_PATH is only used in linking the executable near the end of the compilation.

File: cime_config/acme/machines/config_machines.xml [hunk 4]
```diff
@@ -993,33 +993,100 @@
...
+	   <modules mpilib="openmpi">
+	     <command name="load">openmpi/1.8.3</command>
+	   </modules>
+	   <modules>
+	     <command name="load">netcdf/4.3.2</command>
+	   </modules>
+	 </module_system>
+	 <environment_variables>
+	   <env name="OMP_STACKSIZE">64M</env>
+	   <env name="NETCDF_HOME">$ENV{NETCDF_LIB}/../</env>
```
Comment (singhbalwinder) [inline cime_config/acme/machines/config_machines.xml]: In config_compilers.xml:
`<NETCDF_PATH> $(NETCDF_HOME)</NETCDF_PATH>`
`<SLIBS> -L$(NETCDF_PATH)/lib -lnetcdf -lnetcdff -lpmi -L$(MPI_LIB) -lmpich</SLIBS>`

**Comment by jedwards4b** [comment]
If you look at file config_compilers.xml there is an entry for constance. 
You should replace 
```  <NETCDF_PATH> $(NETCDF_HOME)</NETCDF_PATH>```
with ```  <NETCDF_PATH> $(NETCDF_LIB)/../</NETCDF_PATH>``` and remove the line in config_machines.xml you can do the same with MKL_PATH and avoid setting either of these in the environment.

**Comment by singhbalwinder** [comment]
Thanks @jedwards4b ! That worked!

**Comment by singhbalwinder** [comment]
I have created a new PR with base maint-cime5.1. I am going to close this PR.

**Pull request closed without merging**